infrastructure/infrastructure.py

```
@@ -0,0 +1,476 @@
+
+def ensure_ls_keypair() -> None:
+    """
+    Ensure a Lightsail keypair exists with name LIGHTSAIL_KEY_NAME.
+    If missing, create it and write the private key locally.
+    """
+    if PRIVATE_KEY_PATH.exists():
+        print(f"using existing local private key={PRIVATE_KEY_PATH}")
+        return
+
+    try:
+        ls.get_key_pair(keyPairName=LIGHTSAIL_KEY_NAME)
```

Ruff (0.13.0)

Undefined name `ls`

(F821)